examples/combobox/js/combobox-datepicker.js

```diff
@@ -0,0 +1,842 @@
+
+    case "Tab":
+      this.cancelButtonNode.focus();
+      if (event.shiftKey) {
+        this.nextYearNode.focus();
+      }
+      this.setMessage('');
+      flag = true;
+      break;
+
+    case "Right":
+    case "ArrowRight":
+      this.moveFocusToNextDay();
+      flag = true;
+      break;
+
+    case "Left":
+    case "ArrowLeft":
+      this.moveFocusToPreviousDay();
+      flag = true;
+      break;
+
+    case "Down":
+    case "ArrowDown":
+      this.moveFocusToNextWeek();
+      flag = true;
+      break;
+
+    case "Up":
+    case "ArrowUp":
+      this.moveFocusToPreviousWeek();
+      flag = true;
+      break;
+
+    case "PageUp":
+      if (event.shiftKey) {
+        this.moveToPreviousYear();
+      }
+      else {
+        this.moveToPreviousMonth();
+      }
+      this.setFocusDay();
+      flag = true;
+      break;
+
+    case "PageDown":
+      if (event.shiftKey) {
+        this.moveToNextYear();
+      }
+      else {
+        this.moveToNextMonth();
+      }
+      this.setFocusDay();
+      flag = true;
+      break;
+
+    case "Home":
+      this.moveFocusToFirstDayOfWeek();
+      flag = true;
+      break;
+
+    case "End":
+      this.moveFocusToLastDayOfWeek();
+      flag = true;
+      break;
+  }
+
+  if (flag) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+};
+
+ComboboxDatePicker.prototype.handleDayClick = function (event) {
+
+  if (!this.isDayDisabled(event.currentTarget)) {
+    this.setComboboxDate(event.currentTarget);
+    this.close();
+  }
+
+  event.stopPropagation();
+  event.preventDefault();
+
+};
+
+ComboboxDatePicker.prototype.handleDayFocus = function () {
+  this.setMessage(this.messageCursorKeys);
+};
+
+// Combobox methods
+
+ComboboxDatePicker.prototype.setComboboxDate = function (domNode) {
+
+  var d = this.focusDay;
+
+  if (domNode) {
+    d = this.getDayFromDataDateAttribute(domNode);
+  }
+
+  this.comboboxNode.value = (d.getMonth() + 1) + '/' + d.getDate() + '/' + d.getFullYear();
+
+};
+
+ComboboxDatePicker.prototype.getDateFromCombobox = function () {
+
+  var parts = this.comboboxNode.value.split('/');
+
+  if ((parts.length === 3) &&
+      Number.isInteger(parseInt(parts[0])) &&
+      Number.isInteger(parseInt(parts[1])) &&
+      Number.isInteger(parseInt(parts[2]))) {
+    this.focusDay = new Date(parseInt(parts[2]), parseInt(parts[0]) - 1, parseInt(parts[1]));
+    this.selectedDay = new Date(this.focusDay);
+  }
+  else {
+    // If not a valid date (MM/DD/YY) initialize with todays date
+    this.focusDay = new Date();
+    this.selectedDay = new Date(0,0,1);
+  }
+
+};
+
+ComboboxDatePicker.prototype.setMessage = function (str) {
+
+  function setMessageDelayed () {
+    this.messageNode.textContent = str;
+  }
+
+  if (str !== this.lastMessage) {
+    setTimeout(setMessageDelayed.bind(this), 200);
+    this.lastMessage = str;
+  }
+};
+
+ComboboxDatePicker.prototype.setFocusCombobox = function () {
+  this.comboboxNode.focus();
+};
```

Comment:
No longer needed since @mcking65's notes.

Reply:
Thanks, I removed the unused methods